### PR TITLE
Remove old keys from database

### PR DESCRIPTION
**Pull request by dizider**

### CODE REVIEW

File: internal/functions/efgs/database/database.go [hunk 1]
```diff
@@ -122,6 +122,19 @@ func (db Connection) RemoveDiagnosisKey(keys []*efgsapi.DiagnosisKeyWrapper) err
 	return nil
 }
 
+//RemoveOldKeys Removes keys older than date provided as parameter.
+func (db Connection) RemoveOldKeys(date string) error {
```
Comment (jendakol) [inline internal/functions/efgs/database/database.go]: Please name the parameter as `before_date` or something.

File: internal/functions/efgs/cleanup-database.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	"net/http"
+	"time"
+)
+
+//CleanupDatabase Remove old (more than 14 days) keys from database.
+func CleanupDatabase(w http.ResponseWriter, r *http.Request) {
+	var ctx = r.Context()
+	logger := logging.FromContext(ctx).Named("efgs.CleanupDatabase")
+	now := time.Now()
+	dateFrom := now.AddDate(0, 0, -14).Format("2006-01-02")
```
Comment (jendakol) [inline internal/functions/efgs/cleanup-database.go]: I think it'd be worth it to have a constant `KeyValidityDays = 14` for usage both here and in upload (I don't remember a particular place but it's somewhere there, maybe in `database.go`).

Comment (jendakol) [inline internal/functions/efgs/cleanup-database.go]: I think there's no need to have `now` in a variable here...

**Review by jendakol — CHANGES_REQUESTED** [review]
Only some nitpicks, but a good job!

File: internal/functions/efgs/cleanup-database.go (hunk 2 above)
Comment (dizider) [reply internal/functions/efgs/cleanup-database.go]: I couldn't find the correct const. So I had to create it.

**Review by jendakol — APPROVED** [review]

File: internal/functions/efgs/database/database.go [hunk 3]
```diff
@@ -23,6 +23,9 @@ type Connection struct {
 	inner lazyConnection
 }
 
+//KeyValidityDays Key validity period since creation in days.
+const KeyValidityDays = 14
+
```
Comment (jendakol) [inline internal/functions/efgs/database/database.go]: This is not used at all, is it?

File: internal/functions/efgs/database/database.go [hunk 4]
```diff
@@ -122,6 +125,19 @@ func (db Connection) RemoveDiagnosisKey(keys []*efgsapi.DiagnosisKeyWrapper) err
 	return nil
 }
 
+//RemoveOldKeys Removes keys older than date provided as parameter.
+func (db Connection) RemoveOldKeys(dateFrom string) error {
```
Comment (jendakol) [inline internal/functions/efgs/database/database.go:154]: Sorry, but shouldn't the param be named more like `date_until`, or `older_than_date` or something?

**Review by jendakol — CHANGES_REQUESTED** [review]

**Comment by jendakol** [comment]
BTW please change `KEY_VALIDITY_DAYS` to something less general :wink:

**Review by jendakol — APPROVED** [review]

**Pull request merged**